core/src/main/scala/akka/persistence/jdbc/state/AkkaSerialization.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.state
+
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+import akka.annotation.InternalApi
+import akka.serialization.{ Serialization, Serializers }
+
+import scala.util.Try
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object AkkaSerialization {
+
+  case class AkkaSerialized(serId: Int, serManifest: String, payload: Array[Byte])
+
+  def serialize(serialization: Serialization, payload: Any): Try[AkkaSerialized] = {
+    val p2 = payload.asInstanceOf[AnyRef]
+    val serializer = serialization.findSerializerFor(p2)
+    val serManifest = Serializers.manifestFor(serializer, p2)
+    val serialized = serialization.serialize(p2)
+    serialized.map(payload => AkkaSerialized(serializer.identifier, serManifest, payload))
+  }
+
+  def fromRow(serialization: Serialization)(row: DurableStateTables.DurableStateRow): Try[AnyRef] = {
+    serialization.deserialize(row.statePayload, row.stateSerId, row.stateSerManifest.getOrElse(""))
+  }
```

Comment:
Maybe better to have only one `AkkaSerialization` object. And we can have two `fromRow` methods. Either removing the currying or having explicit named for the `fromRow` method.

This is internal API, so we don't need to be cautious when making bin-incompatible changes.

Reply:
Refactored to have 1 `AkkaSerialization` object 

* Moved `AkkaSerialization` to `akka.persistence.jdbc`. It was in `akka.persistence.jdbc.journal.dao`. Didn't want to have `state` depend on `journal`. Hence the move
* Added a method `fromDurableStateRow` to the object `AkkaSerialization` to be used from `state`. Changed the name to avoid any ambiguity.
* Removed the other `AkkaSerialization` object